### DISCUSSION TITLE
How to extract value from SimpleValue::Float/SimpleValue::Double?

### DISCUSSION THREAD

**Discussion by medigor** [rabbitmq/rabbitmq-stream-rust-client#252] [General]
I don't know how to write this code:
```rust
if let SimpleValue::Float(float) = SimpleValue::Float(1.0.into()) {
    let f: f32 =  ?????;
    println!("{f}");
}
```

**Comment by DanielePalaia** [comment]
Here we do a similar thing for String:

https://github.com/rabbitmq/rabbitmq-stream-rust-client/blob/main/examples/superstreams/send_super_stream.rs#L10

**Reply by medigor** [reply]
Hi!
for Float/Double/Timestamp it doesn't work

![image](https://github.com/user-attachments/assets/271f5643-fdd7-4b46-9f29-010876fbbc72)

**Reply by mypasswordisqwerty** [reply]
Also you can't get value of private protocol::message::amqp::types::MessageId.
